src/Traits/AbstractCommandTrait.php

```diff
@@ -86,7 +88,62 @@ protected function getRemoteSettings($host, $user, $keyFile, $app, $timestamp =
                 $defaults['server-' . $key] = $server;
             }
         }
-        return $this->tokenReplace($this->getConfig()->get('remote'), $replacements) + $defaults;
+        $settings = $this->tokenReplace($this->getConfig()->get('remote'), $replacements) + $defaults;
+        return $this->processEnvironmentOverrides($settings);
+    }
+
+    /**
+     * Process environment-specific overrides.
+     *
+     * @param array $settings
+     * @return array
+     *
+     * @see self::getRemoteSettings
+     */
+    protected function processEnvironmentOverrides($settings)
+    {
+        $defaults = [
```

Comment:
$defaults doesn't need to be reinitialized every single call, so it might be made static, but doesn't have to be IMO.
But, it might be a good idea to declare it outside of the method, this would made possible tests able to override/test it a lot more easily.

Still, not a deal breaker as far as I am concerned ;)